types/ldap_settings.pp

```diff
@@ -5,7 +5,6 @@
     'port'             => Stdlib::Port,
     'base_dn'          => String[1],
     'bind_dn'          => String[1],
-    'bind_dn_password' => Sensitive,
```

Comment:
```
'bind_dn_password' => Optional[Variant[Sensitive[String[1],String[1]]],
```
I think if we do it this way it won't be a breaking change and people are able to set the hash in hiera.